### DISCUSSION TITLE
Glances API status check returns Error 405 - Method Not Allowed

### DISCUSSION THREAD

**Discussion by peevishkiwi** [nicolargo/glances#2826] [Q&A]
I have installed Glances (V4) on a RasPi  in order to monitor it with gethomepage. When starting glances -w, i can access the monitoring with a webbrowser on MYIP:61208 and all values and sensors are shown - so it works. 

When trying add it to gethomepage, the widget shows "API error". I have verified with a curl, that Glances and gethomepage can "talk to each other" with a ping from the container.

Here is the issue: when testing the glances API status with curl -I http://localhost:61208/api/4/status as described in the API documentation, the following error is shown:
![Glances API status error](https://github.com/nicolargo/glances/assets/167554480/b071c4d4-07d9-4a91-8dbb-6ef042fba2b3)
As of the documentation, it should be "HTTP/1.0 200 OK", everything else means there is an issue. This is most likely the reason that gethomepage shows an API error, because it can not get the data from Glances.

I searched the documentation as well as the issues and discussion here on GitHub, but couldn't find anything related to the 405 error.
Does anyone know what i can do to fix the 405 error?

**Comment by RazCrimson** [comment]
Are you sure that you need it to be a HEAD request? 
Would a normal GET not work for gethomepage?

**Comment by peevishkiwi** [comment]
No to be honest i am not sure, no. This was just in the glances documentation to test out if the API is reachable or not, so i tested it. And as gethomepage is showing "API Error" there seems to be something wrong with the API. Couldn't find any other solution or hint online so far.

**Comment by RazCrimson** [comment]
The API not being consistent with the docs has been fixed now. 
Ref: https://github.com/nicolargo/glances/pull/2842

But i don't think that is the issue you are facing.

Could you take a look at: https://gethomepage.dev/latest/more/troubleshooting/#service-widget-errors

Or maybe anything here cab help: https://github.com/nicolargo/glances/issues/2763

**Comment by peevishkiwi** [comment]
Thank you for the fix! I guess it will be included in the next release? Then i will wait for it to test again and tick one possible reason of my list 😄 

I checked the CURL from the gethomepage URL you linked a few days ago and it was successful, so the container had a connection to the glances device. But will try it again after the next release and doublecheck. I am travelling for my company atm, but will take a look at  #2763 once i am back!

**Reply by RazCrimson** [reply]
If you using docker and can access your machine now, you can use the `dev` tagged images which reflect the latest develop branch to test.

The images are listed here:
https://hub.docker.com/r/nicolargo/glances/tags?page=&page_size=&ordering=&name=dev
